csharp/src/Ice/IncomingResponseFrame.cs

```diff
@@ -39,9 +39,12 @@ public ArraySegment<byte> Payload
                         Debug.Assert(ReplyStatus == ReplyStatus.OK || ReplyStatus == ReplyStatus.UserException);
 
                         // TODO: works only when Payload called first before reading anything. Need a better version!
-                        // TODO: not efficient to create an array here
                         // TODO: provide Encoding property
-                        _payload = new ArraySegment<byte>(InputStream.ReadEncapsulation(out Encoding _));
+                        ArraySegment<byte> payload = InputStream.ReadEncapsulation(out Encoding _);
```

Comment:
This code will of course get reworked/simplified once the Incoming frames are refactored.

Reply:
Yes I only  rework ReadEncapsulation to not create a copy